pkg/storage/fs/cback/cback.go

```diff
@@ -0,0 +1,473 @@
+	ctxpkg "github.com/cs3org/reva/pkg/ctx"
+	"github.com/cs3org/reva/pkg/errtypes"
+	"github.com/cs3org/reva/pkg/mime"
+	"github.com/cs3org/reva/pkg/storage"
+	"github.com/cs3org/reva/pkg/storage/fs/registry"
+	"github.com/mitchellh/mapstructure"
+	"github.com/pkg/errors"
+)
+
+type cback struct {
+	conf *Options
+}
+
+func init() {
+	registry.Register("cback", New)
+}
+
+// New returns an implementation to the storage.FS interface that talks to
+// cback
+func New(m map[string]interface{}) (fs storage.FS, err error) {
+
+	c := &Options{}
+	if err = mapstructure.Decode(m, c); err != nil {
+		return nil, errors.Wrap(err, "Error Decoding Configuration")
+	}
+
+	// returns the storage.FS interface
+	return &cback{conf: c}, nil
+
+}
+
+func (fs *cback) GetMD(ctx context.Context, ref *provider.Reference, mdKeys []string) (*provider.ResourceInfo, error) {
+	var ssID, searchPath string
+
+	user, _ := ctxpkg.ContextGetUser(ctx)
+	UID, _ := ctxpkg.ContextGetUserID(ctx)
+
+	resp, err := fs.matchBackups(user.Username, ref.Path)
+
+	if err != nil {
+		fmt.Print(err)
+		return nil, err
+	}
+
+	snapshotList, err := fs.listSnapshots(user.Username, resp.ID)
+
+	if err != nil {
+		fmt.Print(err)
+		return nil, err
+	}
+
+	ssID, searchPath = fs.pathTrimmer(snapshotList, resp)
+
+	//fmt.Printf("The ssID is: %v\nThe Path is %v\n", ssID, searchPath)
+
+	if resp.Source == ref.Path {
+		setTime := v1beta1.Timestamp{
+			Seconds: uint64(time.Now().Unix()),
+			Nanos:   0,
+		}
+
+		ident := provider.ResourceId{
+			OpaqueId:  ref.Path,
+			StorageId: "cback",
+		}
+
+		ri := &provider.ResourceInfo{
+			Etag:          "",
+			PermissionSet: &permID,
+			Checksum:      &checkSum,
+			Mtime:         &setTime,
+			Id:            &ident,
+			Owner:         UID,
+			Type:          provider.ResourceType_RESOURCE_TYPE_CONTAINER,
+			Size:          0,
+			Path:          ref.Path,
+			MimeType:      mime.Detect(true, ref.Path),
+		}
+
+		return ri, nil
+
+	}
+
+	ret, err := fs.statResource(resp.ID, ssID, user.Username, searchPath, resp.Source)
+
+	if err != nil {
+		fmt.Print(err)
+		return nil, err
+	}
+
+	setTime := v1beta1.Timestamp{
+		Seconds: ret.Mtime,
+		Nanos:   0,
+	}
+
+	ident := provider.ResourceId{
+		OpaqueId:  ret.Path,
+		StorageId: "cback",
+	}
+
+	ri := &provider.ResourceInfo{
+		Etag:          "",
+		PermissionSet: &permID,
+		Checksum:      &checkSum,
+		Mtime:         &setTime,
+		Id:            &ident,
+		Owner:         UID,
+		Type:          ret.Type,
+		Size:          ret.Size,
+		Path:          ret.Path,
+	}
+
+	if ret.Type == 2 {
+		ri.MimeType = mime.Detect(true, ret.Path)
+	} else {
+		ri.MimeType = mime.Detect(false, ret.Path)
+	}
+
+	return ri, nil
+}
+
+func (fs *cback) ListFolder(ctx context.Context, ref *provider.Reference, mdKeys []string) ([]*provider.ResourceInfo, error) {
+
+	var path string = ref.GetPath()
+	var ssID, searchPath string
+
+	user, _ := ctxpkg.ContextGetUser(ctx)
+	UID, _ := ctxpkg.ContextGetUserID(ctx)
```

Comment:
same as a previous comment

Reply:
Checked if user is inContext, if not will return UserRequired error.